great_expectations/agent/agent.py

```diff
@@ -212,29 +218,26 @@ def _reject_correlation_id(self, id: str):
             self._correlation_ids.clear()
         return should_reject
 
-    @classmethod
-    def _get_config(cls) -> GXAgentConfig:
+    def _get_config(self) -> GXAgentConfig:
         """Construct GXAgentConfig."""
 
         # ensure we have all required env variables, and provide a useful error if not
-
-        class GxAgentConfigSettings(pydantic.BaseSettings):
-            gx_cloud_base_url: str = "https://api.greatexpectations.io"
-            gx_cloud_organization_id: str
-            gx_cloud_access_token: str
-
         try:
-            config = GxAgentConfigSettings()
+            config = GxAgentServerSessionConfig()
+            self._agent_server_session_config = config
```

Comment:
Feels weird to stick this config on `GXAgent`. Also a bit confusing since there's already a `GXAgentConfig` in `self._config`, which is more concerned with the queue. Maybe we should be more explicit and indicate this somehow?

I think it would make more sense to stick the `GxAgentServerSessionConfig` config inside of `AgentServerSession`. Could make `config` init arg optional and if None, just have this try block.

Also just realizing we're also in a method called `_get_config`... lot's to keep straight in working memory :).